### PR TITLE
Update includes in store on create/update

### PR DESCRIPTION
**Pull request by gavmck**
When running updateEntity or createEntity with includes, the JSON Schema API returns the included objects, but this Redux JSON API doesn't write them back into the store.

I've patched the code to pull the includes down as well as we need this for a project I'm currently working on. All tests passing.

Hope that's OK!

### CODE REVIEW

File: test/jsonapi.js [hunk 1]
```diff
@@ -311,7 +320,7 @@ describe('Creation of new resources', () => {
   it('should add reverse relationship when inserting new resource', () => {
     const updatedState = reducer(state, apiCreated(taskWithTransaction));
 
-    const { data: taskRelationship } = updatedState.transactions.data[0].relationships.task;
+    const taskRelationship = updatedState.transactions.data[0].relationships.task;
 
     expect(taskRelationship.type).toEqual(taskWithTransaction.type);
     expect(taskRelationship.id).toEqual(taskWithTransaction.id);
```
Comment (egeriis) [inline test/jsonapi.js]: This doesn't seem right. Relationships should have a `data` property which points to a resource identifier.

Comment (gavmck) [inline test/jsonapi.js]: Ah yes you are right there! I should have updated the check on taskWithTransaction a couple of lines down. Have just pushed a fix.

I understand the nerves around changes in the tests. Unfortunately the tests used mock data and it was the data I needed to change.

Both `updateEntity()` and `createEntity()` would only pass the `json.data` back from the API and we also need `json.included` so I had to pass the whole `json` variable through as on `readEndpoint()`. This meant the data structure for the tests had to change to mirror the new data structure. This join is not currently tested, maybe we should add some?

**Comment by egeriis** [comment]
Sorry for the radio silence, @gavmck. I'll look into this one of the coming days :)

**Comment by gavmck** [comment]
No worries, thanks!

**Pull request merged**